File: docs/examples/py/src/buckets_change.py
from reduct import Client, BucketSettings, QuotaType




async def change_bucket():
    # Create a client with the base URL and API token
    with Client("http://localhost:8383", api_token="my-token") as client:
        # Get an existing bucket
        bucket = await client.get_bucket("example-bucket")

        # Change the quota size of the bucket to 5GB
        new_settings = BucketSettings(quota_size=5000_000_000)
        await bucket.set_settings(new_settings)

        assert (await bucket.get_settings()).quota_size == 5000_000_000


if __name__ == "__main__":
    import asyncio

    asyncio.run(change_bucket())

File: versioned_docs/version-1.20.x/examples/go/src/buckets_change.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	model "github.com/reductstore/reduct-go/model"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get an existing bucket
	bucket, err := client.GetBucket(context.Background(), "example-bucket")
	if err != nil {
		panic(err)
	}

	// Change the quota size of the bucket to 5GB
	newSettings := model.NewBucketSettingBuilder().
		WithQuotaType(model.QuotaTypeFifo).
		WithQuotaSize(5_000_000_000).
		Build()
	err = bucket.SetSettings(context.Background(), newSettings)
	if err != nil {
		panic(err)
	}

	settings, err := bucket.GetSettings(context.Background())
	if err != nil {
		panic(err)
	}

	if settings.QuotaSize != 5_000_000_000 {
		panic("Bucket settings were not updated correctly")
	}
}
